lib/knock/client.rb

```diff
@@ -6,16 +6,22 @@ module Client
     include Kernel
 
     def client
-      return @client if defined?(@client)
-
-      @client = Net::HTTP.new(Knock::API_HOSTNAME, 443)
-      @client.use_ssl = true
-
-      @client
+      Net::HTTP.new(Knock.config.api_hostname, 443).tap do |http_client|
+        http_client.use_ssl = true
+        http_client.open_timeout = Knock.config.timeout
+        http_client.read_timeout = Knock.config.timeout
+        http_client.write_timeout = Knock.config.timeout if RUBY_VERSION >= '2.6.0'
+      end
     end
 
     def execute_request(request:)
-      response = client.request(request)
+      begin
+        response = client.request(request)
+      rescue Net::OpenTimeout, Net::ReadTimeout, Net::WriteTimeout
+        raise TimeoutError.new(
+          message: 'API Timeout Error',
```

Comment:
Nit: would be good to include the original error here to know what kind of timeout it is.